File: cases/taint/java/map-iteration-positive/MapIterationPositive.java
package dataflowbench.taint;

import java.util.HashMap;
import java.util.Map;

final class MapIterationPositive {
    static int dfb_source() { // DFB-SOURCE: map-iteration-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: map-iteration-sink

    static void run() {
        Map<String, Integer> tainted = new HashMap<>();
        tainted.put("entry", dfb_source()); // DFB-WITNESS: map-iteration-store
        Map<String, Integer> clean = new HashMap<>();
        clean.put("entry", 0);
        for (Map.Entry<String, Integer> entry : tainted.entrySet()) { // DFB-WITNESS: map-iteration-entries
            dfb_sink(entry.getValue());
        }
    }
}

File: cases/taint/rust/map-iteration-positive/map_iteration_positive.rs
use std::collections::HashMap;

fn dfb_source() -> i32 { // DFB-SOURCE: map-iteration-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: map-iteration-sink

fn run() {
    let mut carrier: HashMap<String, i32> = HashMap::new();
    let mut other: HashMap<String, i32> = HashMap::new();
    other.insert(String::from("payload"), 0);
    carrier.insert(String::from("payload"), dfb_source()); // DFB-WITNESS: map-iteration-store
    for (_key, value) in &carrier {
        dfb_sink(*value);
    }
}
